scvi/model/base/_utils.py

```diff
@@ -14,13 +14,84 @@
 
 from scvi.data._constants import _SETUP_METHOD_NAME
 from scvi.data._download import _download
+from scvi.module.base._decorators import _move_data_to_device
 from scvi.utils import track
 
 from ._differential import DifferentialComputation
 
 logger = logging.getLogger(__name__)
 
 
+def subset_distribution(my_distribution, index, dim=0):
+    """Utility function to subset the parameter of a Pytorch distribution."""
+    return my_distribution.__class__(
+        **{
+            name: torch.index_select(
+                getattr(my_distribution, name), dim=dim, index=index
+            )
+            for name in my_distribution.arg_constraints.keys()
+        }
+    )
+
+
+def move_distribution(my_distribution, device):
```

Comment:
Let's use this

https://github.com/pyro-ppl/pyro/blob/9afb089f7f1f32b2bba5ce3064bee2769823285a/pyro/distributions/util.py#L159-L177